cli/install/install.go

```diff
@@ -0,0 +1,800 @@
+			"--depth=1", "-j", "18", repoLink,
+			"--recursive", dst)
+	} else {
+		err = util.ExecuteCommand("git", verbose, logFile, dst, "clone", "-b", tag,
+			"--depth=1", "-j", "18", repoLink,
+			"--recursive", dst)
+	}
+
+	return err
+}
+
+// copyBuildedTT copies tt binary.
+func copyBuildedTT(dst, path, version string, flags InstallationFlag,
+	logFile *os.File) error {
+	var err error
+	if _, err := os.Stat(dst); os.IsNotExist(err) {
+		err = os.Mkdir(dst, FolderRights)
+		if err != nil {
+			return fmt.Errorf("Unable to create bindir: %s", err)
+		}
+	}
+	if flags.Reinstall {
+		err = os.Remove(filepath.Join(dst, version))
+		if err != nil {
+			printLog(logFile.Name())
+			return err
+		}
+	}
+	err = util.CopyFilePreverse(filepath.Join(path, "tt"), filepath.Join(dst, version))
+	return err
+}
+
+// installTt installs selected version of tt.
+func installTt(version string, binDir string, flags InstallationFlag, distfiles string) error {
+
+	// Check binary directory.
+	if binDir == "" {
+		return fmt.Errorf("Install path parameter is empty.")
+	}
+	logFile, err := ioutil.TempFile("", "tarantool_install")
+	if err != nil {
+		return err
+	}
+	defer os.Remove(logFile.Name())
+
+	// Check if that version is already installed.
+	log.Infof("Checking existing...")
+	if checkExisting(version, binDir) && !flags.Reinstall {
+		log.Infof("%s version of tt already exists, updating symlink...", version)
+		err := util.CreateSymLink(version, binDir, "tt", true)
+		log.Infof("Done")
+		return err
+	}
+
+	// Check tt dependencies.
+	if !flags.Force {
+		log.Infof("Checking dependencies...")
+		if !programDependenciesInstalled("tt") {
+			return nil
+		}
+	}
+
+	path, err := os.MkdirTemp("", "tt_install")
+	if err != nil {
+		return err
+	}
+	os.Chmod(path, FolderRights)
+
+	if !flags.Noclean {
+		defer os.RemoveAll(path)
+	}
+
+	// Download tt.
+	if flags.Local {
+		if checkExisting("tt", distfiles) {
+			log.Infof("Local files found, installing from them...")
+			localPath, _ := util.JoinAbspath(distfiles, "tt")
+			err = copy.Copy(localPath, path)
+			if err != nil {
+				printLog(logFile.Name())
+				return err
+			}
+			util.ExecuteCommand("git", flags.Verbose, logFile, path, "checkout", "master")
+		} else {
+			return fmt.Errorf("Can't find distfiles directory.")
+		}
+	} else {
+		log.Infof("Downloading tt...")
+		err = downloadRepo(TTLink, "master", path, logFile, flags.Verbose)
+	}
+
+	if err != nil {
+		printLog(logFile.Name())
+		return err
+	}
+	// Build tt.
+	log.Infof("Building tt...")
+	err = util.ExecuteCommand("mage", flags.Verbose, logFile, path,
+		"build")
+	if err != nil {
+		printLog(logFile.Name())
+		return err
+	}
+
+	// Copy binary.
+	log.Infof("Copying executable...")
+	err = copyBuildedTT(binDir, path, version, flags, logFile)
+	if err != nil {
+		printLog(logFile.Name())
+		return err
+	}
+
+	// Set symlink.
+	err = util.CreateSymLink(version, binDir, "tt", true)
+	if err != nil {
+		printLog(logFile.Name())
```

Comment:
Does CreateSymLink write something to the log file?

Reply:
logFile have logs if -v flag is not set, so we need to print It. e.g previous execute of git wrote logs to log file, we need to print them if there is an error